packages/indexer/native/src/lib.rs

```diff
@@ -488,14 +488,25 @@ declare_types! {
                 _ => return cx.throw_error("io_type should be input or output or both!")
             };
 
+            let skip = cx.argument::<JsValue>(1)?;
+            let skip_number = if skip.is_a::<JsNumber>() {
+                skip.downcast::<JsNumber>().or_throw(&mut cx)?.value() as u64
+            } else {
+                0_u64
+            };
+
             let mut this = cx.this();
             let hashes = {
                 let guard = cx.lock();
                 let mut iterator = this.borrow_mut(&guard);
                 let mut hashes = vec![];
+                let mut current_counter = 0;
```

Comment:
Rust's iterator has a [skip](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.skip) method, can we leverage that?

Reply:
The pattern sounds feasible and more idiomatic. I'll give it a try.